service/src/test/java/bio/terra/workspace/service/resource/controlled/ControlledResourceServiceTest.java

```diff
@@ -181,6 +187,84 @@ private void cleanUpSharedWorkspace() {
         reusableWorkspace.getWorkspaceId(), user.getAuthenticatedRequest());
   }
 
+  @Test
+  @DisabledIfEnvironmentVariable(named = "TEST_ENV", matches = BUFFER_SERVICE_DISABLED_ENVS_REG_EX)
+  void createSasToken() throws Exception {
+    UUID workspaceUuid = workspace.getWorkspaceId();
+    workspaceService.createWorkspace(workspace, user.getAuthenticatedRequest());
+    workspaceService.createAzureCloudContext(
+        workspaceUuid,
+        "job-id-123",
+        user.getAuthenticatedRequest(),
+        null,
+        new AzureCloudContext(
+            "0cb7a640-45a2-4ed6-be9f-63519f86e04b",
+            "3efc5bdf-be0e-44e7-b1d7-c08931e3c16c",
+            "mrg-terra-integration-test-20211118"));
+    StairwayTestUtils.pollUntilComplete(
+        "job-id-123", jobService.getStairway(), Duration.ofSeconds(30), Duration.ofSeconds(300));
+
+    UUID storageAccountId = UUID.randomUUID();
+    ControlledAzureStorageResource storageAccount =
+        new ControlledAzureStorageResource(
+            workspaceUuid,
+            storageAccountId,
+            "sa-" + workspaceUuid.toString(),
+            "",
+            CloningInstructions.COPY_NOTHING,
+            null,
+            PrivateResourceState.NOT_APPLICABLE,
+            AccessScopeType.ACCESS_SCOPE_SHARED,
+            ManagedByType.MANAGED_BY_USER,
+            null,
+            "sa" + storageAccountId.toString().substring(0, 6),
+            "eastus");
+    controlledResourceService.createControlledResourceSync(
+        storageAccount, null, user.getAuthenticatedRequest(), getAzureStorageCreationParameters());
+
+    UUID storageContainerId = UUID.randomUUID();
+    ControlledAzureStorageContainerResource storageContainer =
+        new ControlledAzureStorageContainerResource(
+            workspaceUuid,
+            storageContainerId,
+            "sc-" + workspaceUuid.toString(),
+            "",
+            CloningInstructions.COPY_NOTHING,
+            null,
+            PrivateResourceState.NOT_APPLICABLE,
+            AccessScopeType.ACCESS_SCOPE_SHARED,
+            ManagedByType.MANAGED_BY_USER,
+            null,
+            storageAccount.getResourceId(),
+            "sc-" + storageContainerId);
+    controlledResourceService.createControlledResourceSync(
+        storageContainer,
+        null,
+        user.getAuthenticatedRequest(),
+        getAzureStorageContainerCreationParameters());
+
+    OffsetDateTime startTime = OffsetDateTime.now().minusMinutes(15);
+    OffsetDateTime expiryTime = OffsetDateTime.now().plusHours(1);
+    String sas =
+        controlledResourceService.createAzureStorageContainerSasToken(
+            workspaceUuid,
+            storageContainer.getResourceId(),
+            startTime,
+            expiryTime,
+            user.getAuthenticatedRequest());
+    assertThat("sas is formatted correctly", sas.startsWith("sv"));
```

Comment:
This isn't exactly a comprehensive assertion that we're getting a valid token. Do people think this is fine? Maybe we add in a direct call to Azure to try and use it? This also might be better suited as an integration test than a connected test.

Reply:
Would it be worth then using the sas token to access the storage container and do some basic operation like creating a BlobClient and calling the `exists` method?